### PR TITLE
Add cancel endpoint for conversations

### PR DESCRIPTION
**Pull request by bidisha-c**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [ ] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. If updating an existing version, please select the specific language SDKs and CLIs that must be refreshed after the swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No refresh required for updates in this PR

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist 

> **Applicability**: :warning: 
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs 
> - Adding new properties 
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [ ] Check this box if any of the following appy to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you are using OpenAPIHub to initialize the PR for adding a new version. More details, refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [ ] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If any of the following scenarios apply to the PR, request approval from the Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 
- [ ] Removing API(s) in a stable version
- [ ] Removing properties in a stable version
- [ ] Removing API version(s) in a stable version
- [ ] Updating API in a stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Review by mikaelsitruk — APPROVED** [review]

File: specification/cognitiveservices/data-plane/Language/preview/2022-05-15-preview/analyzeconversations.json [hunk 1]
```diff
@@ -179,6 +179,46 @@
           }
         }
       }
+    },
+    "/analyze-conversations/jobs/{jobId}:cancel": {
```
Comment (heaths) [inline specification/cognitiveservices/data-plane/Language/preview/2022-05-15-preview/analyzeconversations.json:183]: @tg-msft @johanste I don't see guidelines for exactly how to cancel an LRO. Does this follow the pattern we want? It's similar to what I've seen before, but not entirely clear if this is the right pattern.

Also, and to @bidisha-c as well, why would `:cancel` return an Operation-Location header? Presuming canceling takes a while (plausible), is this a new LRO endpoint apart from the original job that was started, or is it the same and the customer can use either to wait for the "cancelled" status?

Comment (heaths) [inline specification/cognitiveservices/data-plane/Language/preview/2022-05-15-preview/analyzeconversations.json:183]: @tg-msft @johanste I don't see guidelines for exactly how to cancel an LRO. Does this follow the pattern we want? It's similar to what I've seen before, but not entirely clear if this is the right pattern.

Also, and to @bidisha-c as well, why would `:cancel` return an Operation-Location header? Presuming canceling takes a while (plausible), is this a new LRO endpoint apart from the original job that was started, or is it the same and the customer can use either to wait for the "cancelled" status?

File: specification/cognitiveservices/data-plane/Language/preview/2022-05-15-preview/analyzeconversations.json [hunk 2]
```diff
@@ -1718,12 +1758,6 @@
     "TranscriptConversationItem": {
       "type": "object",
       "description": "Additional properties for supporting transcript conversation.",
-      "required": [
-        "text",
-        "lexical",
-        "itn",
-        "maskedItn"
-      ],
```
Comment (heaths) [inline specification/cognitiveservices/data-plane/Language/preview/2022-05-15-preview/analyzeconversations.json:1726]: How is this change related, and are these not still required? Even for response objects, we want a list of never-null/missing properties for better code generation. Was this intentional, or just a mistake? I.e. I don't mind unrelated changes in the PR, it just isn't obvious if this was intended.

Comment (heaths) [inline specification/cognitiveservices/data-plane/Language/preview/2022-05-15-preview/analyzeconversations.json:1726]: How is this change related, and are these not still required? Even for response objects, we want a list of never-null/missing properties for better code generation. Was this intentional, or just a mistake? I.e. I don't mind unrelated changes in the PR, it just isn't obvious if this was intended.

File: specification/cognitiveservices/data-plane/Language/preview/2022-05-15-preview/examples/SuccessfulAnalyzeTextJobsCancelRequest.json [hunk 3]
```diff
@@ -8,7 +8,7 @@
   "responses": {
     "202": {
       "headers": {
-        "Operation-Location": "{Endpoint}/language/analyze-text/jobs/{jobId}?api-version=2022-05-01-preview"
+        "Operation-Location": "{Endpoint}/language/analyze-text/jobs/{jobId}?api-version=2022-05-15-preview"
```
Comment (heaths) [inline specification/cognitiveservices/data-plane/Language/preview/2022-05-15-preview/examples/SuccessfulAnalyzeTextJobsCancelRequest.json:11]: More of service feedback, but our new LRO guidelines (internal doc currently) prescribe not sending an `api-version` back in LRO or pageable links. I don't think it's wrong per-se and this wouldn't hold up the feature, but more of an FYI.

Comment (heaths) [inline specification/cognitiveservices/data-plane/Language/preview/2022-05-15-preview/examples/SuccessfulAnalyzeTextJobsCancelRequest.json:11]: More of service feedback, but our new LRO guidelines (internal doc currently) prescribe not sending an `api-version` back in LRO or pageable links. I don't think it's wrong per-se and this wouldn't hold up the feature, but more of an FYI.

Comment (heaths) [reply specification/cognitiveservices/data-plane/Language/preview/2022-05-15-preview/examples/SuccessfulAnalyzeTextJobsCancelRequest.json:11]: Either way, SDKs will add/overwrite it with whatever version devs specifying (default is latest when SDK was built) in their client options when constructing a client.

Comment (heaths) [reply specification/cognitiveservices/data-plane/Language/preview/2022-05-15-preview/examples/SuccessfulAnalyzeTextJobsCancelRequest.json:11]: Either way, SDKs will add/overwrite it with whatever version devs specifying (default is latest when SDK was built) in their client options when constructing a client.

**Review by heaths — CHANGES_REQUESTED** [review]
Just want to get clarification on a question first, but otherwise LGTM. I'll ping the two people offline as well.

File: specification/cognitiveservices/data-plane/Language/preview/2022-05-15-preview/analyzeconversations.json (hunk 2 above)
Comment (bidisha-c) [reply specification/cognitiveservices/data-plane/Language/preview/2022-05-15-preview/analyzeconversations.json:1726]: This change we noticed while testing. The idea is we need one type to be present but not all are required. Is there an open API construct that allows for this?

File: specification/cognitiveservices/data-plane/Language/preview/2022-05-15-preview/analyzeconversations.json (hunk 1 above)
Comment (tg-msft) [reply specification/cognitiveservices/data-plane/Language/preview/2022-05-15-preview/analyzeconversations.json:183]: Yes, that's the typical pattern.  I don't think it's explicitly written down in our guidelines today which we should fix in the new updates.

Comment (heaths) [reply specification/cognitiveservices/data-plane/Language/preview/2022-05-15-preview/analyzeconversations.json:183]: Offline, @tg-msft confirmed this is correct.

Comment (bidisha-c) [reply specification/cognitiveservices/data-plane/Language/preview/2022-05-15-preview/analyzeconversations.json:183]: I checked it is the same for both TA and conversations. The idea seems to be since this is an LRO to have the Operation-Location to enable customers to verify that the job status was canceled.

File: specification/cognitiveservices/data-plane/Language/preview/2022-05-15-preview/examples/SuccessfulAnalyzeTextJobsCancelRequest.json (hunk 3 above)
Comment (bidisha-c) [reply specification/cognitiveservices/data-plane/Language/preview/2022-05-15-preview/examples/SuccessfulAnalyzeTextJobsCancelRequest.json:11]: Ok. Keeping it as is for now but we can address this in the later version.

Comment (bidisha-c) [reply specification/cognitiveservices/data-plane/Language/preview/2022-05-15-preview/examples/SuccessfulAnalyzeTextJobsCancelRequest.json:11]: Keeping it as is for now /

**Review by heaths — APPROVED** [review]

File: specification/cognitiveservices/data-plane/Language/preview/2022-05-15-preview/analyzeconversations.json (hunk 2 above)
Comment (heaths) [reply specification/cognitiveservices/data-plane/Language/preview/2022-05-15-preview/analyzeconversations.json:1726]: No, so this is fine. Thanks for clarifying!

**Comment by heaths** [comment]
@lmazuel everything looks good, but that dup symbol error for python generation is occuring again. Can you merge anyway?

/cc @weshaggard
